lib/phoenix/tracker/delta_generation.ex

```diff
@@ -54,11 +54,16 @@ defmodule Phoenix.Tracker.DeltaGeneration do
     end)
   end
 
-  defp delta_fullfilling_clock(generations, remote_clock) do
+  defp delta_fullfilling_clock(generations, remote_context) do
     generations
     |> Enum.with_index()
-    |> Enum.find(fn {%State{range: {local_start, _local_end}}, _} ->
-      Clock.dominates?(remote_clock, local_start)
+    |> Enum.find(fn {%State{range: {local_start, local_end}}, _} ->
+      if Clock.dominates_or_equal?(remote_context, local_start) and
```

Comment:
No need for the if here. Could be rewritten as:

Clock.dominates_or_equal?(remote_context, local_start) and
   not Clock.dominates?(remote_context, local_end)